### ISSUE TITLE
False positive UnusedVariable in global scope

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#539]
A quick solution would be to have a different issue type for unused global variables?
(I haven't checked if this check is done across files)

UnusedVariable is emitted for the following example.

```php
<?php

$OPTS = ['key' => 'value'];
function example() : void {
    global $OPTS;
    var_export($OPTS);
}
example();
```

**Comment by TysonAndre** [comment]
Also, `@psalm-suppress UnusedVariable` isn't working for me on vimeo's master branch. (Both in function and global scope)

```php
<?php

function test() : void {
    /** @psalm-suppress UnusedVariable */
    $var = 2;
}
```

I expected no errors, but saw this error with --find-dead-code

```
ERROR: UnusedVariable - src/suppress.php:5:5 - Variable $var is never referenced
    /** @psalm-suppress UnusedVariable */
    $var = 2;
```

**Comment by muglug** [comment]
I won't make a new issue, but that should be something that Psalm should pickup. Using the `global` keyword should register as a use for that variable.

**Comment by nickyr** [comment]
Sorry @tysonandre, @muglug can’t fix this tonight because he has to go to bed so he can get up early and go skiing with me tomorrow. 

He feels bad about it, but I insist.
